primitives/benches/crypto_primitives/poseidon_mht.rs

```diff
@@ -0,0 +1,342 @@
+
+    let num_leaves = 2usize.pow(BENCH_HEIGHT as u32 - 1);
+
+    let mut rng = XorShiftRng::seed_from_u64(1231275789u64);
+
+    let mut tree = MNT4PoseidonMHT::init(BENCH_HEIGHT, num_leaves);
+
+    c.bench_function(format!("Batch half Poseidon MHT Eval for MNT4 ({} leaves)", num_leaves).as_str(), move |b| {
+        b.iter(|| {
+            for _ in 0..num_leaves/2 {
+                tree.append(MNT4753Fr::rand(&mut rng));
+            }
+            tree.finalize_in_place();
+            tree.reset();
+        })
+    });
+}
+
+fn batch_poseidon_mht_eval_mnt6_half(c: &mut Criterion) {
+
+    let num_leaves = 2usize.pow(BENCH_HEIGHT as u32 - 1);
+
+    let mut rng = XorShiftRng::seed_from_u64(1231275789u64);
+
+    let mut tree = MNT6PoseidonMHT::init(BENCH_HEIGHT, num_leaves);
+
+    c.bench_function(format!("Batch half Poseidon MHT Eval for MNT6 ({} leaves)", num_leaves).as_str(), move |b| {
+        b.iter(|| {
+            for _ in 0..num_leaves/2 {
+                tree.append(MNT6753Fr::rand(&mut rng));
+            }
+            tree.finalize_in_place();
+            tree.reset();
+        })
+    });
+}
+
+fn batch_poseidon_mht_eval_mnt4_1_4(c: &mut Criterion) {
+
+    let num_leaves = 2usize.pow(BENCH_HEIGHT as u32 - 1);
+
+    let mut rng = XorShiftRng::seed_from_u64(1231275789u64);
+
+    let mut tree = MNT4PoseidonMHT::init(BENCH_HEIGHT, num_leaves);
+
+    c.bench_function(format!("Batch 1/4 Poseidon MHT Eval for MNT4 ({} leaves)", num_leaves).as_str(), move |b| {
+        b.iter(|| {
+            for _ in 0..num_leaves/4 {
+                tree.append(MNT4753Fr::rand(&mut rng));
+            }
+            tree.finalize_in_place();
+            tree.reset();
+        })
+    });
+}
+
+fn batch_poseidon_mht_eval_mnt6_1_4(c: &mut Criterion) {
+
+    let num_leaves = 2usize.pow(BENCH_HEIGHT as u32 - 1);
+
+    let mut rng = XorShiftRng::seed_from_u64(1231275789u64);
+
+    let mut tree = MNT6PoseidonMHT::init(BENCH_HEIGHT, num_leaves);
+
+    c.bench_function(format!("Batch 1/4 Poseidon MHT Eval for MNT6 ({} leaves)", num_leaves).as_str(), move |b| {
+        b.iter(|| {
+            for _ in 0..num_leaves/4 {
+                tree.append(MNT6753Fr::rand(&mut rng));
+            }
+            tree.finalize_in_place();
+            tree.reset();
+        })
+    });
+}
+
+fn batch_poseidon_mht_eval_mnt4_interleaved(c: &mut Criterion) {
+
+    let num_leaves = 2usize.pow(BENCH_HEIGHT as u32 - 1);
+
+    let mut rng = XorShiftRng::seed_from_u64(1231275789u64);
+
+    let mut tree = MNT4PoseidonMHT::init(BENCH_HEIGHT, num_leaves);
+
+    c.bench_function(format!("Batch interleaved Poseidon MHT Eval for MNT4 ({} leaves)", num_leaves).as_str(), move |b| {
+        b.iter(|| {
+            for _ in 0..num_leaves/3 {
+                tree.append(MNT4753Fr::zero());
+            }
+            for _ in 0..(num_leaves * 2)/3 {
+                tree.append(MNT4753Fr::rand(&mut rng));
+            }
```

Comment:
Does this "interleaved" test address that one third of all leaves are "empty"?
As far as I remember the constant for an empty leaf at the bottom of the tree is chosen as zero?

Reply:
I want to test the case where the first third of the leaves are empty and the others not